### DISCUSSION TITLE
xDrip does not recognise new G7 sensor, despite correct code, and it working on Tandem T-slim pump

### DISCUSSION THREAD

**Discussion by shrewsdog** [NightscoutFoundation/xDrip#3687] [Bug reports]
Issue is xDrip not recognising a new sensor (which is working on Tandem T-Slim pump).  System Status has correct TX ID for new sensor, but transmitter days and battery voltages are clearly for the old transmitter as days are 10, and voltage a 280 and b 176. Data source is shown as G6/G7/1/1+ Transmitter, and xDrip version is dfcbe80-20240917.  Phone is Motorola Moto g(6) Play.

Catalogue of errors by me.  Had G7 on day nine.  Had been erratic, so inserted new one early.  Then realised I needed to stop the first one.  Used stop sensor which stopped both (at this point I believed I had not paired sensor 2).  Then removed the 2 sensor TXs from bluetooth list on my phone and rebooted.  Inserted sensor 3, paired it with pump which is receiving data correctly.  New TX ID on phone is correct, but the status page is showing the old day nine sensor and battery data, and the main page records sensor error and 'Ends 7.48' tomorrow.  I opted to put an incorrect TX ID in Xdrip, allowed it to try to connect, and when it failed reinputted the correct TX, but no change.

Both the expiring sensor and its replacement have been sat in microwave for hours.  Applicator code for new sensor is correct. 

It is as though the tx data for the first sensor is 'locked' on the phone.

**Comment by Navid200** [comment]
Please post a screenshot of the status page.

**Reply by Navid200** [reply]
Please leave the issue alone.
Please post the status page screenshot here in the discussion.  To do that, please don't respond to an email.
Please log into GitHub and go to the discussions, respond to the post and tap on the paperclip symbol and choose the image  file you want to attach.

**Comment by shrewsdog** [comment]
![xDrip-Status-Screenshot-2024-09-28-12-15-02 (002)](https://github.com/user-attachments/assets/fd94f41a-24d2-4675-a506-96b7c66112a2)
Latest status update

**Reply by Navid200** [reply]
I don't have any experience with t:slim unfortunately.
Is there a way to temporarily disable the pump from connecting with the sensor?

**Reply by shrewsdog** [reply]
Never done this, but the manual refers to putting pump into storage mode on a long time basis, and it gives instructions to resurrect it.  Will this do, and if so how long should it be off for?

**Reply by Navid200** [reply]
I don't want to possibly mess up the pump operation.  So, I am just looking at options to see what is possibly safe.
This is something that I have been concerned about and have reported it to the developers before.

My thinking is that if t:slim is not connected for 15 minutes, G7 will go into rapid connect mode and will look for connection.  Then, xDrip should be able to connect.  It will need 10-15 more minutes to stablilise.
After that, if enabling the pump will not reconnect it, we will need to disable xDrip collection: https://navid200.github.io/xDrip/docs/Stop-xDrip.html
The pump should reconnect.  
Now that xDrip has connected once, it should be easier for it to establish connectivity when you reenable collection in xDrip.

But, I am not 100% sure that it would be safe to disable the pump.

**Reply by shrewsdog** [reply]
The manual says that to bring the pump out of storage mode, it is only necessary to connect it to a power source.  Disappointingly, it says nothing about existing settings being restored.  I do have a note of my basal rates.  The only other option I can see in the manual is to "Stop G7 sensor", but that is where this problem started.

**Reply by Navid200** [reply]
No, you should not stop.  
Is the pump tubeless?  If it has tubes. can you wrap it with aluminum foil to see if it would temporarily disconnect it from the sensor?

**Reply by shrewsdog** [reply]
First, thank you so much for your time.  Second, the pump is tubed.  I could disconnect the tube, and wrap the pump in tin foil, but could it go in the microwave instead?  I have had my lunch bolus, so should manage being disconnected for a while.  What time scale would you envisage?

**Reply by Navid200** [reply]
After you put the pump in the microwave, I expect that xDrip will connect in 20 minutes.  When that happens, you will need to wait for 2-3 consecutive reads.  This adds up to 35 minutes.

After that, you can bring the pump back out.  Hopefully, it will reconnect.  If it turns out that it doesn't, you can disable collection in xDrip as I explained before.

But, before doing any of this, make sure that you have disabled battery optimization for xDrip and for Bluetooth.
https://navid200.github.io/xDrip/docs/BatteryOpt.html
https://navid200.github.io/xDrip/docs/BluetoothBatteryOpt.html

**Reply by Navid200** [reply]
I am on my way out.  I may or may not be able to respond occasionally for the next few hours.

**Reply by shrewsdog** [reply]
Noted, and thank you

**Reply by shrewsdog** [reply]
Sorry to report no resolution.  Pump was suspended and in microwave for 60 minutes.  It reconnected to the sensor within about 3 minutes.  x-Drip did multiple scans, all timing out after 5 minutes.  Phone was within a foot of the sensor for almost all that time..  Main screen still reporting 'G7 State not currently known'.  Events log show nothing for the last three hours when last entry was 'switching to scan always on due to connect failures metric:0'  As to battery optimisation, x-Drip was already set to not optimised.  I have now reset the bluetooth battery to always optimised.  Screenshot of status page attached, but I think nothing has changed.
![xDrip-Status-Screenshot-2024-09-28-15-14-38 (002)](https://github.com/user-attachments/assets/c37f2564-0203-45ea-9f21-bb11dd2d092d)

**Comment by jamorham** [comment]
Please enable OB1 debugging - https://github.com/NightscoutFoundation/xDrip/wiki/Enabling-Debugging-for-OB1-collector then wait some time for it to gather data of failing to connect then use the event log upload feature and report back here something to provide a clue for the identity given when uploading the logs so I can find which are yours and I will take a look for what is happening.

If you haven't done so already, before doing that you may want to reboot your phone in case something is up with the bluetooth.

**Reply by shrewsdog** [reply]
Thank you so much.  Phone was restarted, verbose mode on, and 30 minute log uploaded with AZAZAZAZ which will hopefully help you locate me.

**Comment by shrewsdog** [comment]
I appreciate we are probably in different time zones, but worry in case my reply on github (which I am not familiar with) was not correctly sent.

 

Reply read - Thank you so much. Phone was restarted, verbose mode on, and 30 minute log uploaded with AZAZAZAZ which will hopefully help you locate me.

**Reply by Navid200** [reply]
You clutter the thread every time you respond to an email in order to post here.
You just have to be patient.  He is a volunteer like the rest of us with family and work and other obligations.
